service/worker/scheduler/workflow.go

```diff
@@ -77,6 +77,10 @@ const (
 	maxListMatchingTimesCount = 1000
 
 	rateLimitedErrorType = "RateLimited"
+
+	// Size of the time generator cache, the generator batches getNextTime function calls and
+	// caches the result to avoid the overhead of SideEffects
+	TimeGeneratorCacheSize = 10
```

Comment:
can be package-private
```suggestion
	timeGeneratorCacheSize = 10
```